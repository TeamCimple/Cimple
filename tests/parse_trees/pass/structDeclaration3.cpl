struct MyStruct {
	int a;
	int b;
	float c;

	MyStruct() {
		a = 1;
		b = 2;
		c = 3.0;
	}
};

int (MyStruct m) help() {
	printf("Hello\n");
	return 0;
}

int (MyStruct m) help2() {
	return m.help();
}

int main() {
	return 0;
}
