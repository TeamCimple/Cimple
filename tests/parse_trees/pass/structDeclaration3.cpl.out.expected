Struct((int Var(a),), (int Var(b),), (float Var(c),), , MyStruct, Constructor(MyStructBody:
        CompoundStatement(Declarations:  StatementList: Statement(EQUALS(a, Int(1))), Statement(EQUALS(b, Int(2))), Statement(EQUALS(c, Float(3.))))),, )FuncDecl(Name: Var(main) ReturnType: int Parameters:  Body: CompoundStatement(Declarations:  StatementList: RETURN(Int(0)))) , FuncDecl(Name: Var(help2) ReturnType: RECEIVER(MyStruct,m) int Parameters:  Body: CompoundStatement(Declarations:  StatementList: RETURN(MethodCall(Receiver: m,Call(FunctionName: help Params: )))) , FuncDecl(Name: Var(help) ReturnType: RECEIVER(MyStruct,m) int Parameters:  Body: CompoundStatement(Declarations:  StatementList: Statement(Call(FunctionName: printf Params: String("Hello\n"))), RETURN(Int(0))))
